content/en/docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -28,40 +28,158 @@ DNS is a built-in Kubernetes service launched automatically
 using the addon manager
 [cluster add-on](http://releases.k8s.io/{{< param "githubbranch" >}}/cluster/addons/README.md).
 
-The running DNS Pod holds 3 containers:
+Following Kubernetes v1.12, CoreDNS is declared the default DNS Server instead of kube-dns in kubeadm. You may want to refer
+to the documentation of other installers to verify which is the default DNS server.
 
-- "`kubedns`": watches the Kubernetes master for changes
-  in Services and Endpoints, and maintains in-memory lookup structures to serve
-  DNS requests.
-- "`dnsmasq`": adds DNS caching to improve performance.
-- "`sidecar`": provides a single health check endpoint
-  to perform healthchecks for `dnsmasq` and `kubedns`.
 
 The DNS Pod is exposed as a Kubernetes Service with a static IP.
+Both the CoreDNS and kube-dns service is named as `kube-dns`. This is done so that there is seamless service when one chooses
+to switch between DNS servers. 
 The kubelet passes DNS to each container with the `--cluster-dns=<dns-service-ip>`
 flag.
 
 DNS names also need domains. You configure the local domain in the kubelet
 with the flag `--cluster-domain=<default-local-domain>`.
 
-The Kubernetes cluster DNS server is based on the
-[SkyDNS](https://github.com/skynetservices/skydns) library. It supports forward
-lookups (A records), service lookups (SRV records), and reverse IP address
-lookups (PTR records).
-
-## Inheriting DNS from the node
+The DNS server supports forward lookups (A records), service lookups (SRV records), and reverse IP address lookups (PTR
+records) as per the [DNS specifications] (https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/)
 
 When running a Pod, kubelet prepends the cluster DNS server and searches
 paths to the node's DNS settings.  If the node is able to resolve DNS names
 specific to the larger environment, Pods should also be able to resolve.
-But see [Known issues](/docs/tasks/administer-cluster/dns-debugging-resolution/#known-issues).
+But see [Known issues](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/#known-issues).
 
 If you don't want this, or if you want a different DNS config for pods, you can
 use the kubelet's `--resolv-conf` flag.  Set this flag to "" to prevent Pods from
 inheriting DNS. Set it to a valid file path to specify a file other than
 `/etc/resolv.conf` for DNS inheritance.
 
-## Configure stub-domain and upstream DNS servers
+## CoreDNS
+
+CoreDNS is a flexible, extensible, authoritative DNS server which directly integrates with the Kubernetes API. 
```

Comment:
I think this is a little too marketing-y for this doc.

I would just say:

CoreDNS is a general-purpose authoritative DNS-server that can serve as cluster DNS, complying with...

Reply:
We already introduce coredns and what it is in another introductory document.  I don't think we need to reintroduce it every time we mention it.  Perhaps we should instead link to that doc from here.